File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import ceil

def solution(progresses, speeds):
    answer = []
    
    n = len(progresses)
    
    deploy_days = []
    
    for i in range(n):
        p, s = progresses[i], speeds[i]
        deploy_days.append(ceil((100 - p) / s))
        
    l, r = 0, 1
    
    while l <= r and r < n:
        if deploy_days[l] < deploy_days[r]:
            answer.append(r - l)
            l = r
        else:
            r += 1
            
    answer.append(r - l)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.kt
import java.util.ArrayDeque
import kotlin.math.ceil

class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        val answer = mutableListOf<Int>()
        val queue = ArrayDeque<Int>()
        var preFeat = 0
        var deployFeat = 0
        
        for (i in 0 until progresses.size) {
            val remainProgress = 100 - progresses[i]
            val deployDay = ceil(remainProgress / speeds[i].toDouble()).toInt()
            queue.add(deployDay)
        }
        
        while (queue.isNotEmpty()) {
            val feat = queue.removeFirst()
            
            if (feat > preFeat) {
                if (deployFeat != 0) {
                    answer.add(deployFeat)
                }
                
                preFeat = feat
                deployFeat = 1
            } else {
                deployFeat += 1
            }
        }
        
        answer.add(deployFeat)
        
        return answer.toIntArray()
    }
}